src/_guides/openapi/specification/v3.1/the-perfect-modern-openapi-workflow.md

```diff
@@ -0,0 +1,706 @@
+        uses: bump-sh/github-action@v1
+        with:
+          doc: <your-doc-id-or-slug>
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.yaml
+```
+
+This keeps the API reference documentation always relevant, and because it's
+been used for contract testing there cannot have been any "drift" between the
+API implementation and the OpenAPI describing it, meaning the reference
+documentation must be correct.
+
+### Deploy Mocks
+
+There are countless API mocking tools out there, many of which work with OpenAPI
+to save the manual effort of updating them every time an API changes, whether
+that is throughout design and development phases, or later as the API evolves.
+
+One such tool is Microcks, a self-hosted mock server with an admin interface and
+easily accessible HTTP endpoints that simulate the API being described in
+OpenAPI. You could log into that admin panel and let it know somebody has
+updated the OpenAPI every now and then, but why not automate that to save time.
+
+Using GitHub Actions to handle that update looks a bit like this:
+
+```yaml
+# .github/workflows/api-mocks.yml
+name: Deploy API mocks
+on:
+  push:
+    branches:
+      - main
+jobs:
+  deploy-mocks:
+    if: ${{ github.event_name == 'push' }}
+    name: Deploy API mocks to Microcks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - uses: microcks/import-github-action@v1
+        with:
+          specificationFiles: 'api/openapi.yaml:true'
+          microcksURL: 'https://mocks.example.com/api/'
+          keycloakClientId:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT }}
+          keycloakClientSecret:  ${{ secrets.MICROCKS_SERVICE_ACCOUNT_CREDENTIALS }}
+```
+
+This workflow could be combined with other jobs, or it could be left as its own
+workflow like this for clarity. Either way, whenever a change is made to the
+`main` branch the mock servers will be updated and instantly reflect the latest
+OpenAPI.
+
+*Learn more about [API mocking with Microcks](_guides/bump-sh-tutorials/mocking-with-microcks.md).*
+
+### Publish SDKs
+
+The quicker a customer can integrate with your API, the quicker your business
+will be making money or solving problems. Some users will be happy to integrate
+directly with the API, but many prefer the ease of working within the
+programming language through Software Development Kits (SDKs).
+
+These can be a lot of work to build and keep up to date, but OpenAPI allows API
+providers to automate the creation of code libraries which allow API consumers
+to work in their programming language of choice instead of poking and prodding
+over HTTP directly. 
+
+SDK generation tools have been around for a long time, but in the past the best
+offering was a cumbersome Java-based open-source tool where you generally had to
+develop your own templates. Modern tooling like
+[Speakeasy](_guides/bump-sh-tutorials/generate-sdks-with-speakeasy.md) allows
```

Comment:
We should also mention Stainless, especially since they have a great guide pending completion and release! https://github.com/bump-sh/docs/pull/336